signac/contrib/job.py

```diff
@@ -104,6 +114,10 @@ def __init__(self, project, statepoint, _id=None):
         # Prepare current working directory for context management
         self._cwd = []
 
+        if statepoint is not None:
+            # Update the project's state point cache immediately if opened by state point
+            self._project._register(self)
```

Comment:
Should this be run directly after line 96?

Reply:
This is fixed in #451.